natural_language_processing/text_generation/alpaca/run.py

```diff
@@ -1,72 +1,50 @@
-import argparse
+import os
 
+import torch
 import transformers
 
 from utils.nlp.alpaca_instruct import AlpacaInstruct
-from utils.pytorch import PyTorchRunner
+from utils.pytorch import PyTorchRunnerV2, apply_compile
 from utils.benchmark import run_model
-from utils.misc import print_goodbye_message_and_die
-
-
-def parse_args():
-    parser = argparse.ArgumentParser(description="Run karpathy/nanoGPT with small modifications.")
-    parser.add_argument("-m", "--model_path",
-                        type=str, required=True,
-                        help="path to the model")
-    parser.add_argument("-b", "--batch_size",
-                        type=int, default=1,
-                        help="batch size to feed the model with")
-    parser.add_argument("-f", "--framework",
-                        type=str, default="pytorch",
-                        choices=["pytorch"],
-                        help="specify the framework in which a model should be run")
-    parser.add_argument("--timeout",
-                        type=float, default=60.0,
-                        help="timeout in seconds")
-    parser.add_argument("--num_runs",
-                        type=int,
-                        help="number of passes through network to execute")
-    parser.add_argument("--dataset_path",
-                        type=str,
-                        help="path to JSON file with instructions")
-    parser.add_argument("--disable_jit_freeze", action='store_true',
-                        help="if true model will be run not in jit freeze mode")
-    return parser.parse_args()
 
 
-def run_pytorch(model_path, batch_size, num_runs, timeout, dataset_path, disable_jit_freeze=False, **kwargs):
+def run_pytorch(model_path, num_runs, timeout, dataset_path, disable_jit_freeze=False, **kwargs):
 
     def run_single_pass(pytorch_runner, dataset):
         inputs = encode(dataset.get_input_array())
-        outputs = pytorch_runner.run(batch_size, inputs=inputs.input_ids, max_new_tokens=100)
+        outputs = pytorch_runner.run(1, inputs=inputs.input_ids, max_new_tokens=100)
         response = decode(outputs[:, inputs.input_ids.shape[1]:])
         dataset.submit_prediction(response)
 
 
     model = transformers.AutoModelForCausalLM.from_pretrained(model_path)
+    model.eval()
+    if os.environ.get("TORCH_COMPILE") == "1":
+        aio = '_aio_profiler_print' in dir(torch._C)
+        model = apply_compile(model, aio)
```

Comment:
maybe let's change that function to torch_compile_maybe() and check for TORCH_COMPILE==1 there?